### PR TITLE
Add URL filter to the securitypolicyviolation event handler

### PR DESCRIPTION
**Pull request by kv-bh**
Discussion originally in https://forum.babylonjs.com/t/38591.
Should fix issue shown in this playground script: https://playground.babylonjs.com/#YMWYF6

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

File: packages/dev/core/src/Misc/fileTools.ts [hunk 1]
```diff
@@ -254,6 +254,8 @@ export const LoadImage = (
     };
 
     const cspHandler = (err: any) => {
+        if (err.blockedURI !== img.src) { return; }
```
Comment (sebavan) [inline packages/dev/core/src/Misc/fileTools.ts:257]: Just wondering if it is always the same here or if it could be converted like absolute / relative ???

Comment (kv-bh) [reply packages/dev/core/src/Misc/fileTools.ts:257]: `blockedURI` should be absolute, and I believe `img.src` is absolute also, at least in the tests I did on some modern browsers. I don't recall if this has been something that has been changed over the years though.

Comment (kv-bh) [reply packages/dev/core/src/Misc/fileTools.ts:257]: I can update it to:
```js
if (err.blockedURI !== new URL(img.src, location.href).toString()) { return; }
```
if you think that would be more robust.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13570/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13570/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13570/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13570/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13570/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13570/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13570/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13570/merge#BCU1XR#0

File: packages/dev/core/src/Misc/fileTools.ts (hunk 1 above)
Comment (kv-bh) [reply packages/dev/core/src/Misc/fileTools.ts:257]: Actually, here's an edge case, which doesn't really make sense to do with a URL fragment character, but could affect other characters if they are escaped, particularly unicode characters:

Adding a fragment to the image URL, e.g. `#test`, will show `#test` in the image's `src` property, but the `blockedURI` will show `%23test`, at least in Chromium-based browsers.

Comment (RaananW) [reply packages/dev/core/src/Misc/fileTools.ts]: They should both be absolute

**Review by RaananW — APPROVED** [review]

File: packages/dev/core/src/Misc/fileTools.ts (hunk 1 above)
Comment (RaananW) [reply packages/dev/core/src/Misc/fileTools.ts]: @sebavan - can I resolve this? or is there any other concern here?

**Pull request merged**